src/components/radios/template.test.js

```diff
@@ -0,0 +1,256 @@
+      expect($secondLabel.attr('for')).toEqual('custom-2')
+    })
+
+    it('render disabled', () => {
+      const { $ } = render('radios', {
+        name: 'example-name',
+        items: [
+          {
+            value: 'yes',
+            text: 'Yes',
+            disabled: true
+          },
+          {
+            value: 'no',
+            text: 'No',
+            disabled: true
+          }
+        ]
+      })
+
+      const $component = $('.govuk-c-radios')
+
+      const $firstInput = $component.find('.govuk-c-radios__item:first-child input')
+      expect($firstInput.attr('disabled')).toEqual('disabled')
+
+      const $secondInput = $component.find('.govuk-c-radios__item:last-child input')
+      expect($secondInput.attr('disabled')).toEqual('disabled')
+    })
+
+    it('render checked', () => {
+      const { $ } = render('radios', {
+        name: 'example-name',
+        items: [
+          {
+            value: 'yes',
+            text: 'Yes'
+          },
+          {
+            value: 'no',
+            text: 'No',
+            checked: true
+          }
+        ]
+      })
+
+      const $component = $('.govuk-c-radios')
+      const $secondInput = $component.find('.govuk-c-radios__item:last-child input')
+      expect($secondInput.attr('checked')).toEqual('checked')
+    })
+
+    it('render item label with html, custom class and attributes', () => {
+      const { $ } = render('radios', {
+        name: 'example-name',
+        items: [
+          {
+            value: 'yes',
+            html: '<b>Yes</b>',
+            label: {
+              attributes: {
+                'data-attribute': 'value',
+                'data-second-attribute': 'second-value'
+              }
+            }
+          },
+          {
+            value: 'no',
+            text: '<b>No</b>',
+            checked: true
+          }
+        ]
+      })
+
+      const $component = $('.govuk-c-radios')
+
+      const $firstLabel = $component.find('.govuk-c-radios__item:first-child label')
+      expect($firstLabel.html()).toContain('<b>Yes</b>')
+      expect($firstLabel.hasClass('govuk-c-radios__label')).toBeTruthy()
+      expect($firstLabel.attr('data-attribute')).toEqual('value')
+      expect($firstLabel.attr('data-second-attribute')).toEqual('second-value')
+
+      const $secondLabel = $component.find('.govuk-c-radios__item:last-child label')
+      expect($secondLabel.hasClass('govuk-c-radios__label')).toBeTruthy()
+      expect($secondLabel.html()).toContain('&lt;b&gt;No&lt;/b&gt;')
+    })
+  })
+
+  it('render extreme fieldset', () => {
+    const { $ } = render('radios', examples['with-extreme-fieldset'])
+
+    const $component = $('.govuk-c-radios')
+    expect($component.find('.govuk-c-error-message').text()).toContain('Please select an option')
+
+    const $fieldset = $('.govuk-c-fieldset')
+    expect($fieldset.hasClass('app-c-fieldset--custom-modifier')).toBeTruthy()
```

Comment:
Lots of assertions here since we just want to know that the fieldset plays nice inside this component. Feels over the top to do individual test blocks (which should be done in the fieldset component tests), maybe a snapshot test would be better here?

Reply:
See the same for label and error message.